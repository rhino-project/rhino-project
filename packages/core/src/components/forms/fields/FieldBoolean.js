import PropTypes from 'prop-types';
import { Input } from 'reactstrap';
import { useController } from 'react-hook-form';
import { useFieldInheritedProps } from 'rhino/hooks/form';
import { useGlobalComponent } from 'rhino/hooks/overrides';

export const FieldBooleanBase = (props) => {
  const { extractedProps, inheritedProps } = useFieldInheritedProps(props);
  const { readOnly, path } = props;
  const {
    field: { ref, value, ...fieldProps },
    fieldState: { error }
  } = useController({
    name: path
  });

  // FIXME: Cannot guarantee that id will be unique
  return (
    <Input
      {...extractedProps}
      {...fieldProps}
      innerRef={ref}
      type="checkbox"
      invalid={!!error}
      checked={value}
      disabled={readOnly}
      {...inheritedProps}
    />
  );
};

FieldBooleanBase.propTypes = {
  path: PropTypes.string.isRequired
};

const FieldBoolean = (props) =>
  useGlobalComponent('FieldBoolean', FieldBooleanBase, props);

export default FieldBoolean;
