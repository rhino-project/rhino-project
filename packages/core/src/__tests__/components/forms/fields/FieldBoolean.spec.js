import FieldBoolean from 'rhino/components/forms/fields/FieldBoolean';
import { sharedFieldTests } from './sharedFieldTests';

describe('FieldBoolean', () => {
  sharedFieldTests(FieldBoolean);
});
