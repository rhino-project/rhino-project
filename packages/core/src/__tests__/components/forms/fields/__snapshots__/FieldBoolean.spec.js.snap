// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldBoolean > render with place holder 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="form-check-input"
    id="dummy"
    name="dummy"
    path="dummy"
    placeholder="placeholder"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`FieldBoolean > renders inside of model context 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="form-check-input"
    id="dummy"
    name="dummy"
    path="dummy"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`FieldBoolean > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
